### PR TITLE
Upgrade plugin to eslint 3.0.1 and fix lint issues

### PR DESCRIPTION
**Pull request by kassens**
Most fixes using `eslint --fix` and one manual line break.

Inspired by the upgrade on the root directory by @zpao:
https://github.com/facebook/relay/commit/cc04adef211017656e77ed765744f0b7b647b70d

### CODE REVIEW

**Comment by kassens** [comment]
@facebook-github-bot import

**Comment by zpao** [comment]
You can probably just remove the `scripts/` entry in the root `.eslintignore` file and then remove all eslint related things from this subpackage. Then you don't have to duplicate anything and have eslint & co installed twice

**Comment by facebook-github-bot-3** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1700373373560946/int_phab) to review.

File: scripts/babel-relay-plugin/lib/RelayQLAST.js [hunk 1]
```diff
@@ -145,7 +145,8 @@ var RelayQLDefinition = function (_RelayQLNode) {
   _createClass(RelayQLDefinition, [{
     key: 'getName',
     value: function getName() {
-      return this.ast.name ? this.ast.name.value : this.getType().getName({ modifiers: false }); // TODO: this.context.definitionName;
+      // TODO: this.context.definitionName;
```
Comment (zpao) [inline scripts/babel-relay-plugin/lib/RelayQLAST.js]: Did you intend to move this comment or was that a re-printing issue (I've seen that happen before so just checking).

Comment (kassens) [inline scripts/babel-relay-plugin/lib/RelayQLAST.js]: `/lib/` is the build output,  don't care where the comment is.

**Pull request closed without merging**